### PR TITLE
added list all conversations endpoint

### LINKED ISSUE #69: Get all conversations
Hello!
Is there a way to get a list of all conversations? Is the only way to get all converstaions through users and admins?

Thanks in advance

### LINKED ISSUE #113: Conversation model field "conversation_parts" is always null despite it shouldn't
I have a problem regarding API conversation model: on your reference I could see this is the Object that contains conversation_parts Object
Now when I try to get the Conversation object from JSON I always get the conversation_parts object with null value despite it has the value when using the Postman request
conversation ID as example
**15673691858**
image
that's how conversation_parts object does look like on Postman response
image
and on the project it's null

I know that on your SDK model conversation_parts field is not object, but List<ConversationPart>, however it didn't work as well.

### PR DESCRIPTION
**Pull request by kmossco**
## What changed?

Added a new endpoint that allows to list **all** conversations for an app, either without passing any parameters or by passing parameters. This closes #69 

Added to the model the conversation rating model that was added to the API recently, and fixed a problem with loading conversation parts. This closes #113

### CODE REVIEW

**Comment by SeanSmithers** [comment]
Can tests be added for these changes?

**Comment by kmossco** [comment]
@SeanSmithers added tests for both not sending the parameters, as well as not sending the right ones. 👍

File: src/Intercom/Clients/ConversationsClient.cs [hunk 1]
```diff
@@ -52,5 +52,31 @@ public Conversation View(String id, bool? displayAsPlainText = null)
...
+            if (parameters == null)
+            {
+                throw new ArgumentNullException(nameof(parameters));
+            }
+
+            if (!parameters.ContainsKey("order") &&
+                !parameters.ContainsKey("sort") &&
+                !parameters.ContainsKey("before"))
+            {
+                throw new ArgumentException(String.Format("'parameters' argument should include at least order, or sort, or before parameter."));
```
Comment (khalilovcmd) [inline src/Intercom/Clients/ConversationsClient.cs]: does it actually need to?

Comment (kmossco) [reply src/Intercom/Clients/ConversationsClient.cs]: Oh, you're right! I misread our docs 🤦‍♂️ Going to change that!

**Comment by kmossco** [comment]
@khalilovcmd I removed that artificial limitation, now you can pass whatever parameters are useful. Can I get 👀on this again?

**Pull request merged**